Secrets

73VW opened the pull request:
Avancement dans le rapport

73VW committed:
Résolution conflits

73VW committed:
Continuation du projet

greut commented on source/secrets/index.rst:
```diff
@@ -25,13 +25,16 @@ Ce module donne accès à la source d'aléatoire la plus sécurisée que votre s
-Si k,l ou m ne sont pas renseignés, un nombre raisonnable sera utilisé par défaut.
+Si *k*, *l* ou *m* ne sont pas renseignés, un nombre raisonnable sera utilisé par défaut.
+
+Pour résister à une attaque de `force brute`_, les jetons doivent être suffisament longs pour êtres suffisament aléatoires. Cependant, cette notion de suffisance reste assez vague. Plus les ordinateurs deviennent puissants, plus les jetons devront être longs afin de ralonger le temps nécessaire à une machine pour le découvrir.
```
_être_

greut commented on source/secrets/index.rst:
```diff
@@ -40,6 +43,49 @@ Autre
+    		password = ''.join(choice(alphabet) for i in range(10))
+    		if (any(c.islower() for c in password)
+            		and any(c.isupper() for c in password)
+            		and sum(c.isdigit() for c in password) >= 3):
```
Je vais mettre cette ligne dans le CP.

greut commented on source/secrets/index.rst:
```diff
@@ -25,13 +25,16 @@ Ce module donne accès à la source d'aléatoire la plus sécurisée que votre s
-- :py:func:`secrets.token_hex(l) <secrets.token_hex()>` retourne une chaine de texte héxadécimale composée de l bytes.
-- :py:func:`secrets.token_urlsafe(m) <secrets.token_urlsafe()>` retourne une chaine de texte de m bytes utilisable dans une URL_.
+- :py:func:`secrets.token_bytes(k) <secrets.token_bytes()>` retourne une chaine binaire composé de *k* bytes. 
+- :py:func:`secrets.token_hex(l) <secrets.token_hex()>` retourne une chaine de texte héxadécimale composée de *l* bytes convertis chacun en deux digits hexadécimaux.
```
_h**e**xadécimale_

greut commented on source/secrets/index.rst:
```diff
@@ -25,13 +25,16 @@ Ce module donne accès à la source d'aléatoire la plus sécurisée que votre s
 Jetons
 ******
-Ce module fournit également des fonctions pour générer des jetons sécurisés, adaptés à des applications telles que des réinitialisations de mots de passe, des URL difficiles à deviner et similaires.
+Ce module fournit également des fonctions pour générer des jetons sécurisés, adaptés à des applications telles que des réinitialisations de mots de passe via des URL_ difficiles à deviner, des authentifications via API_ tierces, et d'autres utilisations similaires.
```
Juste.

Signer en encrypter de manière symmétrique requierent un secret de bonne qualité.

https://en.wikipedia.org/wiki/Random_number_generator_attack

73VW replied:
> Signer en encrypter de manière symmétrique

Vous entendez quoi par là? Je suis pas sûr de comprendre

greut replied:
Pardon... _Signer **et** encrypter_ 

HMAC pour signer ou AES pour encrypter (AES, c'est `pycrypto` @Skogarmadr)

73VW replied:
Ah d'accord. Mais par contre là ça dépasse vraiment mon domaine de compétences, il faut que je me renseigne un peu plus. Mais merci

greut replied:
HMAC résoud un problème d'authenticité. Est-ce que ce contenu n'a pas été modifié entre temps. Par exemple le navigateur Opera avait un fichier JS corrigeant internet pour eux. Ce fichier est signé et ne sera plus exécuté s'il devait être modifié https://github.com/operasoftware/browserjs/blob/master/desktop/browserjs-12.50.js

https://pymotw.com/3/hmac/index.html

73VW committed:
Corrections fautes d'orthographe

greut commented on source/secrets/index.rst:
```diff
@@ -40,6 +48,52 @@ Autre
+    		if (any(c.islower() for c in password)
+            	and any(c.isupper() for c in password)
+            	and sum(c.isdigit() for c in password) >= 3):
+        	break
```
l'identation me parait bizarre.

73VW replied:
Je sais, elle l'est.
Dans le prochain commit elle sera juste, c'est déjà modifié en local.

greut commented:
Votre éditeur utilise des tabulations au lieu des quatres espaces habituels.

http://pep8.org/#indentation

73VW committed:
Continuation du projet

Correction indenation, tabulation, ajout de l'origine de l'image dans sa
propriété target

73VW committed:
Résolution soucis d'indentation apparu lors de la conversion (oublié de vérifier avant de commit...)

73VW commented:
@greut pour les tabulations c'était un soucis de config dans sublime text, ça devrait être bon maintenant

greut commented:
J'ai sorti l'exemple: 026df83c1e61e981c0d8410680cdc13e1f0bdaf7

73VW commented:
Merci d'avoir sorti l'exemple
